### ISSUE TITLE
wrong beam slope with dotted notes 

### ISSUE THREAD

**Issue by rettinghaus** [rism-ch/verovio#1541]
The stem shortening on edge cases results with dotted notes in wrong beam slopes. This should never happen and beams should at least be horizontal.

![beams_1XX_001](https://user-images.githubusercontent.com/7693447/87062625-96e25480-c20d-11ea-96c6-301f77164c6d.png)

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.1/mei-CMN.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.1/mei-CMN.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.1">
  <meiHead>
    <fileDesc>
      <titleStmt>
        <title>Short stems with fractional beams</title>
      </titleStmt>
      <pubStmt>
        <respStmt>
          <persName>Klaus Rettinghaus</persName>
        </respStmt>
        <date isodate="2020">2020</date>
      </pubStmt>
      <notesStmt>
        <annot>Beams should never contradict the direction of the spanned interval.</annot>
      </notesStmt>
    </fileDesc>
    <encodingDesc>
      <appInfo>
        <application version="3.0.0" label="0">
          <name>Verovio</name>
        </application>
      </appInfo>
    </encodingDesc>
  </meiHead>
  <music>
    <body>
      <mdiv>
        <score>
          <scoreDef>
            <staffGrp>
              <staffDef n="1" lines="5" clef.line="2" clef.shape="G"/>
            </staffGrp>
          </scoreDef>
          <section>
            <measure>
              <staff n="1">
                <layer n="1">
                  <beam>
                    <note dots="1" dur="8" oct="4" pname="f" stem.dir="up"/>
                    <note dur="16" oct="4" pname="e" stem.dir="up"/>
                  </beam>
                  <beam>
                    <note dots="1" dur="8" oct="5" pname="f" stem.dir="up"/>
                    <note dur="16" oct="5" pname="e" stem.dir="up"/>
                  </beam>
                  <beam>
                    <note dots="1" dur="8" oct="4" pname="e" stem.dir="down"/>
                    <note dur="16" oct="4" pname="f" stem.dir="down"/>
                  </beam>
                  <beam>
                    <note dots="1" dur="8" oct="5" pname="e" stem.dir="down"/>
                    <note dur="16" oct="5" pname="f" stem.dir="down"/>
                  </beam>
                </layer>
              </staff>
            </measure>
          </section>
        </score>
      </mdiv>
    </body>
  </music>
</mei>
```

**Comment by craigsapp** [comment]
Here is a possible related example:


<img width="201" alt="Screen Shot 2020-07-19 at 10 21 54 PM" src="https://user-images.githubusercontent.com/3487289/87902458-7fc21480-ca0e-11ea-8b11-d52d3516747e.png">

The second beam is not acceptable, particularly because the beam goes down while the melodic interval goes up.  Also the beam goes down by a third when the melodic interval goes up by a step (related to issue https://github.com/rism-ch/verovio/issues/1557).  Adding augmentation dots to the first note in the beamed group does not alter the rendering of the beam.

MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2020-07-19T22:21:58" version="3.0.0-dev-f59fa44">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mdiv-0000001013787044">
    <score xml:id="score-0000000493271736">
     <scoreDef xml:id="scoredef-0000000636347792">
      <staffGrp xml:id="staffgrp-0000002048861130">
       <staffDef xml:id="staffdef-0000001251762518" n="1" lines="5">
        <clef xml:id="clef-0000000313657087" shape="G" line="2" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <measure xml:id="measure-L1">
       <staff xml:id="staff-0000000478995721" n="1">
        <layer xml:id="layer-L1F1N1" n="1">
         <beam xml:id="beam-L2F1-L3F1">
          <note xml:id="note-L2F1" dur="8" oct="6" pname="c" accid.ges="n" />
          <note xml:id="note-L3F1" dur="16" oct="6" pname="d" accid.ges="n" />
         </beam>
         <beam xml:id="beam-L4F1-L5F1">
          <note xml:id="note-L4F1" dur="16" oct="6" pname="c" accid.ges="n" />
          <note xml:id="note-L5F1" dur="32" oct="6" pname="d" accid.ges="n" />
         </beam>
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

**Comment by rettinghaus** [comment]
Fixed with #1710.
